crypto/evp/pmeth_fn.c

```diff
@@ -13,105 +13,450 @@
 #include <openssl/evp.h>
 #include "internal/cryptlib.h"
 #include "internal/evp_int.h"
+#include "internal/provider.h"
 #include "evp_locl.h"
 
-int EVP_PKEY_sign_init(EVP_PKEY_CTX *ctx)
+static EVP_SIGNATURE *evp_signature_new(OSSL_PROVIDER *prov)
```

Comment:
Looks like we will end up with quite a few of these near identical factory methods...
Could possibly be refactored at some point..

Reply:
I agree to a point...  there's a point where "refactoring" becomes a pile of magic that's hard to follow.  Of course *documenting the magic* (which we are still seriously bad at) might help making it less magic.